### PR TITLE
Added command API, split release function to more methods

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: project_info/release.py [hunk 1]
```diff
@@ -71,15 +130,15 @@ def compact(seq):
     return tuple(item for item in seq if item is not None)
 
 
-def release(project_dir, release_type='patch', files=['pip.json'], next_branch='next', origin='origin'):
-
+def release(release_type='patch', build_hash=None, files=['version.json'], next_branch='next', origin='origin',
```
Comment (matllubos) [inline project_info/release.py]: tenhle command přijde ještě předělat, ale v jiném pr

File: project_info/bin/project-info.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+import sys
+import argparse
```
Comment (radimsuckr) [inline project_info/bin/project-info.py]: Dangling import.

**Comment by radimsuckr** [comment]
Are you sure that all the options should really be options? Aren't some mandatory? If that's the case, then they should be `click.argument` rather than `click.option`.

**Comment by radimsuckr** [comment]
Also it'd be nice to add a tiny bit of documentation, it makes the CLI help much more helpful (and pretty, of course). https://click.palletsprojects.com/en/7.x/documentation/

**Comment by matllubos** [comment]
@radimsuckr argument does not have hepl text. Most of my options have default value.

maybe for release type we can use argument but it has default value too

**Comment by matllubos** [comment]
And yes documentation is good idea

File: project_info/bin/release-bitbucket.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    pass
+
+
+@cli.command()
+@click.option('--username', help='username', type=str, required=True)
+@click.option('--password', help='password', type=str, required=True)
+@click.option('--source_branch_name', help='source git branch name', type=str)
+@click.option('--destination_branch_name', help='destination git branch name', type=str, required=True, default='next')
+@click.option('--repository_name', help='git repository name', type=str, required=True)
+def create_merge_release_pull_request(username, password, source_branch_name, destination_branch_name, repository_name):
```
Comment (matllubos) [inline project_info/bin/release-bitbucket.py]: helper for bitbucket. Maybe we can use similar in the standard deployment. If find way how get title from jira issue :)

File: project_info/bin/release-git.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
```
Comment (matllubos) [inline project_info/bin/release-git.py]: I hope it is all required git commands for the deployment

File: project_info/git_utils.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+
+def create_deployment_branch(environment, remote_name=None, branch_name=None):
+    repo = git.Repo(os.getcwd())
+    g = repo.git
+    source_branch_name = repo.head.reference
+    deployment_branch_name = 'deploy-{}'.format(environment)
+
+    g.checkout('HEAD', b=deployment_branch_name)
+    g.commit('--allow-empty', message='Deployment of "{}"'.format(source_branch_name))
```
Comment (matllubos) [inline project_info/git_utils.py]: I add release branch name to the message. It is easier to get the branch name for the merge to the next/master

File: project_info/types.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+from enum import Enum
+
+from click import Choice
+
+
+class EnumType(Choice):
```
Comment (matllubos) [inline project_info/types.py]: I found it in the click library issues

File: setup.py [hunk 7]
```diff
@@ -14,8 +14,15 @@
...
-        'gitpython>=0.3.2.RC1',
+        'django>=1.11',
+        'gitpython>=2.1.12',
+        'click==7.0',
+        'requests>=2.22.0',
+    ],
+    scripts=[
+        'project_info/bin/release-version.py',
+        'project_info/bin/release-git.py',
+        'project_info/bin/release-bitbucket.py',
```
Comment (matllubos) [inline setup.py]: @radimsuckr your command for ecs should have separate file

File: project_info/bitbucket_utils.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        'destination': {
+            'branch': {
+                'name': destination_branch_name
+            }
+        }
+    }
+    resp = requests.post(url, headers=headers, json=json_data, auth=HTTPBasicAuth(username, password))
+    if resp.status_code != 201:
+        raise UsageError('Bitbucket error: {}'.format(resp.content.decode('utf-8')))
+    return resp.json()['links']['html']['href']
```
Comment (radimsuckr) [inline project_info/bitbucket_utils.py]: Missing newline at the end.

File: setup.py (hunk 7 above)
Comment (radimsuckr) [reply setup.py]: Yeah, I've already separated it. :+1:

**Pull request merged**